### PR TITLE
Add stop method to controller

### PR DESCRIPTION
**Pull request by tfussell**
Required for https://github.com/giantswarm/kvm-operator/pull/1132 so we can start and stop controllers dynamically to watch WC resources from an MC operator.

## Checklist

- [x] Update changelog in CHANGELOG.md.
- [ ] Update roadmap in ROADMAP.md.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,7 +6,7 @@ require (
 	github.com/getsentry/sentry-go v0.9.0
 	github.com/giantswarm/apiextensions/v3 v3.16.1
 	github.com/giantswarm/backoff v0.2.0
-	github.com/giantswarm/exporterkit v0.2.0
+	github.com/giantswarm/exporterkit v0.2.1-0.20210204214054-ebd41525c595
```
Comment (tfussell) [inline go.mod:9]: Will use released version when https://github.com/giantswarm/exporterkit/pull/31 is merged

File: pkg/controller/controller.go [hunk 2]
```diff
@@ -280,6 +282,11 @@ func NewSelector(matchesFunc func(labels labels.Labels) bool) selector.Selector
 	return selector.NewSelector(matchesFunc)
 }
 
+func (c *Controller) Stop(ctx context.Context) {
+	c.collector.Stop(ctx)
+	close(c.stopped)
```
Comment (tfussell) [inline pkg/controller/controller.go:287]: Closing the channel causes the manager to stop reconciling.

File: pkg/controller/controller.go [hunk 3]
```diff
@@ -497,10 +505,10 @@ func (c *Controller) reconcile(ctx context.Context, req reconcile.Request, obj i
 	}
 
 	if m.GetDeletionTimestamp() != nil {
-		event := "delete"
+		eventName := "delete"
```
Comment (tfussell) [inline pkg/controller/controller.go:508]: Unrelated change. My editor complained that the variable name matched an imported package name.

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Added
+
+- Add stop method to stop controller reconciliation and metrics collection.
```
Comment (kopiczko) [inline CHANGELOG.md:14]:
```suggestion
- Add `Controller.Stop` method to stop controller reconciliation and metrics collection.
```

**Review by kopiczko — APPROVED** [review]
LGTM

**Review by fiunchinho — APPROVED** [review]
LGTM

**Comment by tfussell** [comment]
Exporterkit PR has been merged. Will update this PR accordingly once it has been released.

**Review by yulianedyalkova — APPROVED** [review]
